Mathlib/Algebra/Homology/EulerCharacteristic.lean

```diff
@@ -0,0 +1,34 @@
+/-
+Copyright (c) 2025 Jesse Alama. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jesse Alama
+-/
+import Mathlib.Algebra.Homology.HomologicalComplex
+import Mathlib.Algebra.Category.ModuleCat.Basic
+import Mathlib.LinearAlgebra.Dimension.Finrank
+
+/-!
+# Euler characteristic of chain complexes
+
+This file defines the Euler characteristic of a chain complex.
+
+## Main definitions
+
+* `ChainComplex.eulerChar`: The Euler characteristic of a chain complex up to index `n`,
+  defined as the alternating sum `∑_{k=0}^n (-1)^k rank(C_k)` of the ranks of the chain groups.
+
+-/
+
+namespace ChainComplex
+
+variable {R : Type*} [Ring R]
+
+/-- The Euler characteristic of a chain complex up to index `n`, defined as the alternating sum
+`∑_{k=0}^n (-1)^k rank(C_k)` of the ranks of the chain groups from index 0 to n.
+This is most commonly used when the base ring is a field (where rank means dimension)
+or a principal ideal domain. For a chain complex that is zero above some index,
+choosing `n` to be at least that index gives the total Euler characteristic. -/
+noncomputable def eulerChar (C : ChainComplex (ModuleCat R) ℕ) (n : ℕ) : ℤ :=
+  ∑ k ∈ Finset.range (n + 1), (-1 : ℤ)^k * Module.finrank R (C.X k)
```

Comment:
```suggestion
noncomputable def eulerChar (C : ChainComplex (ModuleCat R) ℕ) (n : ℕ) : ℤ :=
  ∑ k ∈ Finset.range n, (-1 : ℤ)^k * Module.finrank R (C.X k)
```
In order to do inductions, it may be easier to start from the `0` instead of `1`. Could you add lemmas `eulerChar_zero`, `eulerChar_succ`?

Reply:
I've added an unbounded Euler characteristic and added a theorem that relates the bounded and unbounded versions. Thanks!